package-res/resources/web/pentaho/type/config/ConfigurationService.js

```diff
@@ -21,13 +21,150 @@ define([
 ], function(Base, SortedList, O, error) {
   "use strict";
 
+  /**
+   * List of names of context variables that are handled "generically" when sorting rules.
+   * More specific first.
+   *
+   * @type {string[]}
+   * @see pentaho.spec.IContextVars
+   * @see _ruleComparer
+   * @see _ruleFilterer
+   */
   var _selectCriteria = [
     "user",
     "theme",
     "locale",
     "application"
   ];
 
+  /**
+   * Map of merge operation name to operation handler function.
+   *
+   * @type {Object.<string, function>}
+   * @see mergeSpecsOne
+   */
+  var _mergeHandlers = {
+    "replace": mergeSpecsOperReplace,
+    "merge": mergeSpecsOperMerge,
+    "add": mergeSpecsOperAdd
+  };
+
+  /**
+   * The ordinal value of the next rule that is registered.
+   *
+   * This is used as the fallback rule order.
+   * Ensures sorting algorithm stability (as insertion order would be lost on resorts).
+   *
+   * @type {number}
+   * @see pentaho.type.config.ConfigurationService#addRule
+   */
+  var _ruleCounter = 0;
+
+  /**
+   * @classDesc The `ConfigurationService` class is the base implementation of
+   * the {@link pentaho.type.IConfigurationService} interface.
+   *
+   * @class
+   * @alias ConfigurationService
+   * @memberOf pentaho.type.config
+   * @amd pentaho/type/config/ConfigurationService
+   *
+   * @extends pentaho.lang.Base
+   * @implements pentaho.type.IConfigurationService
+   *
+   * @description Creates a configuration service instance with no registrations.
+   */
+  var ConfigurationService = Base.extend("pentaho.type.config.ConfigurationService",
+  /** @lends pentaho.type.config.ConfigurationService# */{
+
+    constructor: function() {
+      /**
+       * A map from value type absolute id to
+       * the applicable type configuration rules,
+       * ordered from least to most specific.
+       *
```

Comment:
Suggested:
A map connecting a value type's absolute id to the applicable type configuration rules, ordered from least to most specific.